### ISSUE TITLE
sheet to json node - time issues

### ISSUE THREAD

**Issue by BeniReydman** [sakai-to/node-red-contrib-spreadsheet-in#6]
I have a csv file that is formatted like this:

```
1. Timestamp,Humidity,Pressure,PM1,PM2_5,PM10,Temperature,DateTime
2. 2020-07-09T20:00:07.000Z,37.57400131,992.5411987,16.66666667,21,21.66666667,35.34999847,1594324807.00
3. ...
```

To read the spreadsheet I do the following:
![image](https://user-images.githubusercontent.com/15524115/88409916-3c173400-cd9b-11ea-98b4-4567e8ff3626.png)

The problem is when it is converted, the object I get has an unexpected output of:
```
Timestamp: "7/9/20"
Humidity: "37.57400131"
Pressure: "992.5411987"
PM1: "16.66666667"
PM2_5: "21"
PM10: "21.66666667"
Temperature: "35.34999847"
DateTime: "1594324807.00"
```

I cannot figure out why the Timestamp is always adjusted from `"2020-07-09T20:00:07.000Z"` to` "7/9/20"`? I am using DateTime to try and overcome the issue currently, but it is causing me more problems. Do you happen to know where I went wrong or if this is a bug?

Just for added information, here is the properties of my sheet-to-json node:
![image](https://user-images.githubusercontent.com/15524115/88410551-23f3e480-cd9c-11ea-8b0a-375f0b8474e5.png)

**Comment by sakai-to** [comment]
@BeniReydman Thank you for reporting this issue. I could reproduce the issue with the latest version (0.5.0). I will investigate it as soon as possible.

**Comment by sakai-to** [comment]
@BeniReydman The Timestamp value is adjusted by the parsing function `XLSX.read()` in book node. The value `2020-07-09T20:00:07.000Z` seems to be stored as the number `44022.20841435185` with a number format of `m/d/yy` and a formatted text `7/9/20`. The number format `m/d/yy` is used because the it is default date format for `XLSX.read()` and a csv file has no date format info unlike an Excel file.

I am thinking about how to resolve this issue. So, can you provide me with your expected output on your case? Also, please let me know if there is anything you need to consider.

**Comment by BeniReydman** [comment]
I apologize for the late reply. The expected outcome for my case would be to have the same time format of which I originally had in the csv file. Though, I believe that as long as there is no data loss even if there is a format change then it can be considered a solution. This could mean hard coding the time format if that is the current fastest solution in which can be built upon further in later versions.

**Comment by sakai-to** [comment]
@BeniReydman I found the following description at [Parsing Options on SheetJS documentation](https://docs.sheetjs.com/#parsing-options):

> Excel aggressively tries to interpret values from CSV and other plain text. This leads to surprising behavior! The raw option suppresses value parsing.

So, if the raw option of `XLSX.read()` in book node is set to true, we will be able to get the exact strings written in CSV file. In this case (your CSV file), you will get the following output from sheet-to-json node:

```
Timestamp: "2020-07-09T20:00:07.000Z"
Humidity: "37.57400131"
Pressure: "992.5411987"
PM1: "16.66666667"
PM2_5: "21"
PM10: "21.66666667"
Temperature: "35.34999847"
DateTime: "1594324807.00"
```

Is this your expected output? If so, I will add a raw option to a book node.

**Comment by BeniReydman** [comment]
That would be a nice option to have and for my case it would be the expected output. Thank you.

**Comment by sakai-to** [comment]
@BeniReydman I have added the option to a book node. If you check 'Plain text parsing will not parse values' in the configuration of your book node, then you will get the expected output from your sheet-to-json node. You may additionally select 'Raw values' from 'Data type' in the configuration of your sheet-to-json node. Could you please check the new book node?

**Comment by BeniReydman** [comment]
I apologize, I am not well versed in Node-RED at all and have been using palette's to install nodes onto my project. I don't know how to get the node onto my project from your GitHub. I don't see any changes to the project, where do I access these changes?

**Comment by sakai-to** [comment]
@BeniReydman Sorry for my late reply and your inconvenience. I have published the new version 0.6.0, which includes the enhancement for this issue. Please upgrade node-red-contrib-spreadsheet-in on your Node-RED and check the new book. If you have any questions or anything, please do not hesitate to reopen this issue. Thank you for your cooperation!
